src/sphinx/reference/adapters.rst

```diff
@@ -54,18 +54,221 @@ For persisting the stream processing progress, an application should store the l
+   .. includecode:: ../../../eventuate-example-vertx/src/main/java/com/rbmhtechnology/docs/vertx/japi/Documentation.java
+      :snippet: vertx-publish-producer
+
+Event delivery is always performed with *At-Most-Once* delivery semantics, so no guarantees about the successful delivery of events can be made.
+
+Applications consume events by registering an event bus handler at the configured endpoints.
+
+.. tabbed-code::
+   .. includecode:: ../../../eventuate-example-vertx/src/main/scala/com/rbmhtechnology/docs/vertx/Documentation.scala
+      :snippet: event-processing-vertx-producer
+   .. includecode:: ../../../eventuate-example-vertx/src/main/java/com/rbmhtechnology/docs/vertx/japi/Documentation.java
+      :snippet: event-processing-vertx-producer
+
+Replication progress from the source event log is tracked by persisting the ``localSequenceNr`` of the latest delivered event to the ``StorageProvider`` supplied to the ``VertxAdapter``. After delivery of one or multiple events the replication progress is persisted. The producer continues delivery of events from the latest known ``localSequenceNr`` once the it is started.
+
+Vert.x Point-to-Point Event Producer
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A *Point-to-Point Event Producer* sends an event to a *single* subscriber on the event bus. If a single subscriber is registered for an endpoint all events are delivered to this subscriber. If multiple subscribers are registered for the same endpoint, events are delivered alternately to only one of those subscribers using a non-strict round-robin algorithm. Event routing can be enabled by supplying a partial function.
+
+.. tabbed-code::
+   .. includecode:: ../../../eventuate-example-vertx/src/main/scala/com/rbmhtechnology/docs/vertx/Documentation.scala
+      :snippet: vertx-ptp-producer-at-most-once
+   .. includecode:: ../../../eventuate-example-vertx/src/main/java/com/rbmhtechnology/docs/vertx/japi/Documentation.java
+      :snippet: vertx-ptp-producer-at-most-once
+
+Point-to-point event producers support both *At-Most-Once* and *At-Least-Once* delivery semantics. If not specified otherwise *At-Most-Once* delivery is chosen. *At-Least-Once* delivery is enabled by configuring the adapter accordingly.
+
+.. tabbed-code::
+   .. includecode:: ../../../eventuate-example-vertx/src/main/scala/com/rbmhtechnology/docs/vertx/Documentation.scala
+      :snippet: vertx-ptp-producer-at-least-once
+   .. includecode:: ../../../eventuate-example-vertx/src/main/java/com/rbmhtechnology/docs/vertx/japi/Documentation.java
+      :snippet: vertx-ptp-producer-at-least-once
+
+Events sent by a point-to-point event producer are received by registering an event handler on the event bus.
+
+Using *At-Least-Once* delivery semantics, every event must be confirmed by the receiver. Unconfirmed events are redelivered until a confirmation was received by the adapter. Event handlers signal event confirmation by replying to the event bus message with an arbitrary value.
```

Comment:
The adapter does not process the returned value, so the handler can send any value. 
We could define a case object `ACK` which can be sent? ok for you?